packages/zoe/src/contracts/priceAggregatorChainlink.js

```diff
@@ -0,0 +1,743 @@
+// @ts-check
+
+import { E } from '@agoric/eventual-send';
+import { Far } from '@agoric/marshal';
+import { makeNotifierKit } from '@agoric/notifier';
+import makeStore from '@agoric/store';
+import { Nat, isNat } from '@agoric/nat';
+import { AmountMath } from '@agoric/ertp';
+import { assert, details as X } from '@agoric/assert';
+import {
+  calculateMedian,
+  natSafeMath,
+  makeOnewayPriceAuthorityKit,
+} from '../contractSupport';
+
+import '../../tools/types';
+
+const { add, subtract, multiply, floorDivide, ceilDivide, isGTE } = natSafeMath;
+
+/**
+ * This contract aggregates price values from a set of oracleStatuses and provides a
+ * PriceAuthority for their median.
+ *
+ * @type {ContractStartFn}
+ */
+const start = async zcf => {
+  const {
+    timer: rawTimer,
+    brands: { In: brandIn, Out: brandOut },
+    maxSubmissionCount,
+    minSubmissionCount,
+    restartDelay,
+    timeout,
+    minSubmissionValue,
+    maxSubmissionValue,
+
+    unitAmountIn = AmountMath.make(1n, brandIn),
+  } = zcf.getTerms();
+
+  const unitIn = AmountMath.getValue(unitAmountIn, brandIn);
+
+  /** @type {TimerService} */
+  const timer = rawTimer;
+
+  /** @type {IssuerRecord & { mint: ERef<Mint> }} */
+  let quoteKit;
+
+  /** @type {PriceAuthority} */
+  let priceAuthority;
+
+  /** @type {number} */
```

Comment:
This is actually a BigInt.

```suggestion
  /** @type {bigint} */
```